### PR TITLE
Refactor & Fix all warnings

### PR DESCRIPTION
**Pull request by DesWurstes**
This pull requests resolves **all** warnings issued during compilation by -Wall/-Wextra of the source code and the tests.

---

#497 is needed to enable unittests, as GCC 5.5 poorly supports AVX512. That pull request shows build failure, because there are too many warnings that made into the source code silently, which is the reason new warnings come regularly. Therefore first merging *this* then #497 would prevent new warnings from coming silently. (Though I'll rebase #497 after this one is merged, so please only merge this for now.)

**TL;DR** without this pull request, unit tests won't run properly on Travis!

---

I believe it'd be easier to review it [commit-by-commit](https://github.com/OpenVisualCloud/SVT-AV1/pull/511/commits). I've documented all changes. Forgive me for the mess below, there are actually five commits. I know that it's big pain to review 26 files changed, so let me know if there's something I can do to make it easier for you. 

The last repush only changes the commit date.

### CODE REVIEW

**Comment by EwoutH** [comment]
@DesWurstes What's the status on this PR? Is it ready to be merged?

@hassount @joelsole Have you looked at it?

Possibly related to the errors in #614.

**Comment by DesWurstes** [comment]
Rebased.

> Possibly related to the errors in #614.

Yes, in #497 I did a similar thing.

**Comment by EwoutH** [comment]
@DesWurstes On GCC 8.3.0 I still get the following warning:

```
[ 81%] Built target COMMON_C_DEFAULT
/home/ewoutterhoeven/SVT-AV1/Source/Lib/Common/Codec/EbAdaptiveMotionVectorPrediction.c: In function ‘setup_ref_mv_list’:
/home/ewoutterhoeven/SVT-AV1/Source/Lib/Common/Codec/EbAdaptiveMotionVectorPrediction.c:1120:59: warning: array subscript 2 is above array bounds of ‘IntMv[2]’ {aka ‘union IntMv[2]’} [-Warray-bounds]
                     comp_list[comp_idx][idx] = ref_id[idx][list_idx];
```
Could you look into it?

**Comment by DesWurstes** [comment]
[I can't reproduce from the preprocessed version of the file](https://godbolt.org/z/P_D4mr).
